src/main/java/com/minecolonies/coremod/colony/requestsystem/resolvers/WarehouseRequestResolver.java

```diff
@@ -0,0 +1,163 @@
+import com.google.common.reflect.TypeToken;
+import com.minecolonies.api.colony.requestsystem.IRequestManager;
+import com.minecolonies.api.colony.requestsystem.location.ILocation;
+import com.minecolonies.api.colony.requestsystem.request.IRequest;
+import com.minecolonies.api.colony.requestsystem.requestable.Delivery;
+import com.minecolonies.api.colony.requestsystem.requester.IRequester;
+import com.minecolonies.api.colony.requestsystem.token.IToken;
+import com.minecolonies.api.util.ItemStackUtils;
+import com.minecolonies.coremod.tileentities.TileEntityWareHouse;
+import net.minecraft.item.ItemStack;
+import net.minecraft.tileentity.TileEntity;
+import net.minecraft.util.math.BlockPos;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.List;
+import java.util.UUID;
+
+/**
+ * ----------------------- Not Documented Object ---------------------
+ */
+public class WarehouseRequestResolver extends AbstractRequestResolver<ItemStack>
+{
+
+    public WarehouseRequestResolver(
+                                     @NotNull final ILocation location,
+                                     @NotNull final IToken token)
+    {
+        super(location, token);
+    }
+
+    @Override
+    public Class<? extends ItemStack> getRequestType()
+    {
+        return ItemStack.class;
+    }
+
+    @Override
+    public boolean canResolve(@NotNull final IRequestManager manager, final IRequest<ItemStack> requestToCheck)
+    {
+        if (ItemStackUtils.isEmpty(requestToCheck.getRequest()))
+        {
+            return false;
+        }
+
+        final TileEntity tileEntity = manager.getColony().getWorld().getTileEntity(getLocation().getInDimensionLocation());
+
+        if (tileEntity instanceof TileEntityWareHouse)
+        {
+            final TileEntityWareHouse wareHouse = (TileEntityWareHouse) tileEntity;
+
+            return wareHouse.isInHut(requestToCheck.getRequest());
+        }
+
+        return false;
+    }
+
+    @Nullable
+    @Override
+    public List<IToken> attemptResolve(
+                                        @NotNull final IRequestManager manager, @NotNull final IRequest<ItemStack> request)
+    {
+        final TileEntity tileEntity = manager.getColony().getWorld().getTileEntity(getLocation().getInDimensionLocation());
+
+        if (tileEntity instanceof TileEntityWareHouse)
+        {
+            final TileEntityWareHouse wareHouse = (TileEntityWareHouse) tileEntity;
+            final BlockPos pos = wareHouse.getPositionOfChestWithItemStack(request.getRequest());
+
+            request.setResult(request.getRequest().copy());
+            return Lists.newArrayList(manager.createRequest(new WarehouseChestDeliveryRequester(this, manager.getFactoryController().getNewInstance(UUID.randomUUID(),
+              new TypeToken<IToken>() {}), manager.getFactoryController().getNewInstance(pos, new TypeToken<ILocation>() {}), request.getToken()),
```

Comment:
@Asherslab for now please make a "suppress" tag over the method for now, with the id of the sonar rule and the explanation why we ignore it.